apps/omg_watcher/lib/web/serializers/response.ex

```diff
@@ -19,61 +19,65 @@ defmodule OMG.Watcher.Web.Serializer.Response do
 
   @type response_result_t :: :success | :error
 
-  @spec serialize(map(), response_result_t()) :: %{result: response_result_t(), data: map()}
-  def serialize(data, result) do
-    %{
-      result: result,
-      data: data
-    }
-  end
+  @doc """
+  Append result of operation to the response data forming standard api response structure
+  """
+  @spec serialize(any(), response_result_t()) :: %{result: response_result_t(), data: map()}
+  def serialize(data, result)
+  def serialize(data, :success), do: data |> clean_artifacts() |> to_response(:success)
+  def serialize(data, :error), do: data |> to_response(:error)
 
-  @spec encode16(list(map()) | map(), list(String.t() | atom())) :: map()
-  def encode16(list, fields) when is_list(list) do
-    list |> Enum.map(&encode16(&1, fields))
+  defp to_response(data, result), do: %{result: result, data: data}
+
+  @doc """
+  Removes or encodes fields in response that cannot be serialized to api response.
+  By default, it:
+   * encodes to hex all binary values
+   * removes unloaded ecto associations values
+   * removes metadata fields
+  """
+  def clean_artifacts(reponse) do
```

Comment:
typo, and why you created clean_artifacts? 
clean_value can be public

Reply:
I wanted for it more descriptive name - clean_value is recursive. Typo fixed.